hydromt_sfincs/subgrid.py

```diff
@@ -18,13 +18,200 @@
+        self.z_zmax[active_z] = self.ds["z_zmax"].values.flatten()
+        self.z_volmax[active_z] = self.ds["z_volmax"].values.flatten()
+        for ilevel in range(self.nlevels):
+            self.z_level[ilevel, active_z[0], active_z[1]] = self.ds["z_level"][
+                ilevel
+            ].values.flatten()
+
+        # now use index_mu1 and index_nu1 to put the values of the active cells in the new dataset
+        var_list = ["zmin", "zmax", "ffit", "navg"]
+        for var in var_list:
+            uv_var = self.ds["uv_" + var].values.flatten()
+            self.u_zmin[active_u] = uv_var[index_mu1[active_indices_u]]
+            self.v_zmin[active_v] = uv_var[index_nu1[active_indices_v]]
+
+        var_list_levels = ["havg", "nrep", "pwet"]
+        for var in var_list_levels:
+            for ilevel in range(self.nlevels):
+                uv_var = self.ds["uv_" + var][ilevel].values.flatten()
+                self.u_havg[ilevel, active_u[0], active_u[1]] = uv_var[
+                    index_mu1[active_indices_u]
+                ]
+                self.v_havg[ilevel, active_v[0], active_v[1]] = uv_var[
+                    index_nu1[active_indices_v]
+                ]
+
+    # new way of writing netcdf subgrid tables
+    def write(self, file_name, mask):
+        """Write subgrid table to netcdf file for a regular grid with given mask.
+        Values are only written for active cells (mask > 0)."""
+
+        ds = self.to_xarray(dims=mask.raster.dims, coords=mask.raster.coords)
+
+        # Need to transpose to match the FORTRAN convention in SFINCS
+        ds = ds.transpose("levels", "x", "y")
+
+        # find indices of active cells
+        index_nm, index_mu1, index_nu1 = utils.find_uv_indices(mask)
+
+        # get number of levels
+        nlevels = self.nlevels
+
+        active_cells = index_nm > -1
+        active_indices = np.where(active_cells)[0]
+
+        # Make a new xarray dataset where we only keep the values of the active cells (index_nm > -1)
+        z_zmin = ds["z_zmin"].values.flatten()[active_cells]
+        z_zmax = ds["z_zmax"].values.flatten()[active_cells]
+        z_volmax = ds["z_volmax"].values.flatten()[active_cells]
+        z_level = np.array(
+            [
+                ds["z_level"][ilevel].values.flatten()[active_cells]
+                for ilevel in range(nlevels)
+            ]
+        )
+
+        # get nr of active points (where index_nm > -1)
+        nr_points = max(index_mu1.max(), index_nu1.max()) + 1
+
+        var_list = ["zmin", "zmax", "ffit", "navg"]
+        for var in var_list:
+            uv_var = np.zeros(nr_points)
+            uv_var[index_mu1[active_indices]] = ds["u_" + var].values.flatten()[
+                active_cells
+            ]
+            uv_var[index_nu1[active_indices]] = ds["v_" + var].values.flatten()[
+                active_cells
+            ]
+            locals()["uv_" + var] = uv_var
```

Comment:
Why not storing it in `ds_new` directly? You would need to initialize `ds_new` earlier but I think that makes it more readable. I also suggest using f-strings to improve the readability e.g. 

`ds_new[f"uv_{var}"] = xr.DataArray(uv_var, dims=("npuv"))`

Reply:
Good one, changed this!